.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: Test xen-api
+
+on: [push, pull_request]
+
+jobs:
+  python-test:
+    name: Python tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        test: ["", "-3"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run python tests
+        run: bash ./.travis-python-nosetests${{ matrix.test }}.sh
+
+  ocaml-test:
+    name: Ocaml tests
+    runs-on: ubuntu-20.04
+    env:
+      package: "xapi"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Pull configuration from xs-opam
+        run: |
+          curl https://raw.githubusercontent.com/psafont/xs-opam/gh-act/tools/xs-opam-ci.env 2> /dev/null | cut -f2 -d " " > .env
```

Comment:
This url needs to be changed to point to xapi-project/xs-opam/master when ready

Reply:
I would suggest to to use curl flags 
* `--fail` to detect failure
* `--silent` to avoid other output